### PR TITLE
DFIB-108: Add ESlint into Zed

### PR DESCRIPTION
**Pull request by zamlely**
Ticket: https://spryker.atlassian.net/browse/DFIB-110
CR: https://spryker.atlassian.net/browse/DFIB-111

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,9 @@
     "core-js": "~2.5.5",
     "css-loader": "~2.1.1",
     "es6-promise": "~4.0.5",
+    "eslint": "^6.8.0",
     "file-loader": "~1.1.11",
+    "glob": "^7.1.6",
```
Comment (supproduction) [inline package.json]: Suggest, use `fast-glob` or even `globby`, cause it faster than `glob`.

Both have similar syntax.

File: lib/eslint.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        configFile: settings.paths.esLintConfig,
+        useEslintrc: false,
+        ignorePattern: ['node_modules/']
+    });
+    const formatter = cli.getFormatter('compact');
+    const report = cli.executeOnFiles(pathPattern);
+    console.log(formatter(report.results));
+}).catch(error => {
+    console.error(error);
+});
```
Comment (supproduction) [inline lib/eslint.js]: With `globby` you can simplify this code 

```
const runESLint = async () => {
   const cli = new CLIEngine({
        configFile: settings.paths.esLintConfig,
        useEslintrc: false,
        ignorePattern: ['node_modules/']
    });
    const pathPatternEco = `${settings.paths.ecoDir}/**/*.js`;
   const pathPattern = [`${settings.paths.bundlesDir}/**/*.js`];
    const filesToVerify = await globby([pathPatternEco, pathPattern]);

    const formatter = cli.getFormatter('compact');
    const report = cli.executeOnFiles(filesToVerify);
    console.log(formatter(report.results));
}

runESLint();
```

**Review by supproduction — CHANGES_REQUESTED** [review]

**Review by supproduction — APPROVED** [review]

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        "document": "readonly",
+        "navigator": "readonly",
+        "window": "readonly",
+        "$": "readonly"
+    },
+    "parserOptions": {
+        "ecmaVersion": 2018
+    },
+    // define the rules to override default rules list
+    "rules": {
```
Comment (tolerants) [inline .eslintrc.js:28]: But why it's empty?

**Review by tolerants — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 3 above)
Comment (zamlely) [reply .eslintrc.js:28]: Basic rules are defined in `"extends": "eslint:recommended",`.

Comment (zamlely) [reply .eslintrc.js:28]: Fixed.

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+            "arrays": "never",
+            "objects": "never",
+            "imports": "never",
+            "exports": "never",
+            "functions": "never"
+        }],
+        "comma-spacing": ["error", { "before": false, "after": true }],
+        "comma-style": ["error", "last"],
+        "constructor-super": "error",
+        "curly": ["error"],
```
Comment (danilmoro3) [inline .eslintrc.js]: Please remove this rule

Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion
```

**Review by danilmoro3 — CHANGES_REQUESTED** [review]

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+                ["==", "!=", "===", "!==", ">", ">=", "<", "<="],
+                ["in", "instanceof"]
+            ]
+        }],
+        "no-mixed-spaces-and-tabs": "error",
+        "no-multi-spaces": "error",
+        "no-multi-str": "error",
+        "no-multiple-empty-lines": ["error", { "max": 1, "maxEOF": 0 }],
+        "no-negated-in-lhs": "error",
+        "no-new": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion
```

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,230 @@
+module.exports = {
+    "env": {
+        "commonjs": true,
+        "es6": true,
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

File: .eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+        "swal": "readonly",
+        "DEV": "readonly"
+    },
+    "parserOptions": {
+        "ecmaVersion": 2018
+    },
+    // define the rules to override default rules list
+    "rules": {
+        "accessor-pairs": "error",
+        "arrow-spacing": ["error", { "before": true, "after": true }],
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+        "comma-dangle": ["error", {
+            "arrays": "never",
+            "objects": "never",
+            "imports": "never",
+            "exports": "never",
+            "functions": "never"
+        }],
+        "comma-spacing": ["error", { "before": false, "after": true }],
+        "comma-style": ["error", "last"],
+        "constructor-super": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+        "dot-location": ["error", "property"],
+        "eol-last": ["error", "always"],
+        "eqeqeq": ["error", "always", { "null": "ignore" }],
+        "func-call-spacing": ["error", "never"],
+        "generator-star-spacing": ["error", { "before": false, "after": true }],
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+            "after": true
+        }],
+        "new-cap": ["error", {
+            "newIsCap": true,
+            "capIsNew": false
+        }],
+        "new-parens": "error",
+        "no-array-constructor": "error",
+        "no-caller": "error",
+        "no-class-assign": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+        "no-compare-neg-zero": "error",
+        "no-cond-assign": ["error", "always"],
+        "no-const-assign": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+        "no-constant-condition": ["error", { "checkLoops": false }],
+        "no-control-regex": "error",
+        "no-debugger": "error",
+        "no-delete-var": "error",
+        "no-dupe-args": "error",
+        "no-dupe-class-members": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+        "no-mixed-spaces-and-tabs": "error",
+        "no-multi-spaces": "error",
+        "no-multi-str": "error",
+        "no-multiple-empty-lines": ["error", { "max": 1, "maxEOF": 0 }],
+        "no-negated-in-lhs": "error",
+        "no-new": "error",
+        "no-new-func": "error",
+        "no-new-object": "error",
+        "no-new-require": "error",
+        "no-new-symbol": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+        "no-return-assign": ["error", "except-parens"],
+        "no-return-await": "error",
+        "no-self-assign": "error",
+        "no-self-compare": "error",
+        "no-sequences": "error",
+        "no-shadow-restricted-names": "error",
+        "no-sparse-arrays": "error",
+        "no-tabs": "error",
+        "no-template-curly-in-string": "error",
+        "no-this-before-super": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+            "vars": "all",
+            "args": "all"
+        }],
+        "no-use-before-define": ["error", {
+            "functions": false,
+            "classes": false,
+            "variables": false
+        }],
+        "no-useless-call": "error",
+        "no-useless-computed-key": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+        "no-useless-constructor": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+        "no-useless-escape": "error",
+        "no-useless-rename": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+            "blocks": "never",
+            "switches": "never",
+            "classes": "never"
+        }],
+        "prefer-promise-reject-errors": "error",
+        "quotes": ["error", "single", {
+            "avoidEscape": true,
+            "allowTemplateLiterals": true
+        }],
+        "rest-spread-spacing": ["error", "never"],
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+            "line": {
+                "markers": ["*package", "!", "/", ",", "="]
+            },
+            "block": {
+                "balanced": true,
+                "markers": ["*package", "!", ",", ":", "::", "flow-include"],
+                "exceptions": ["*"]
+            }
+        }],
+        "symbol-description": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+        "template-curly-spacing": ["error", "never"],
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+        "template-tag-spacing": ["error", "never"],
+        "unicode-bom": ["error", "never"],
+        "use-isnan": "error",
+        "valid-typeof": ["error", {
+            "requireStringLiterals": true
+        }],
+        "wrap-iife": ["error", "any", {
+            "functionPrototypeMethods": true
+        }],
+        "yield-star-spacing": ["error", "both"],
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es6 rule

File: .eslintrc.js [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+        "SprykerAjaxCallbacks": "readonly",
+        "confirm": "readonly",
+        "alert": "readonly",
+        "localStorage": "readonly",
+        "antelope": "readonly",
+        "Event": "readonly",
+        "CustomEvent": "readonly",
+        "Image": "readonly",
+        "swal": "readonly",
+        "DEV": "readonly"
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion
        "SprykerAjaxCallbacks": "readonly"
```

File: .eslintrc.js [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+    },
+    "parserOptions": {
+        "ecmaVersion": 2018
+    },
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: es5 uses by default

**Review by danilmoro3 — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 6 above)
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion
        "es6": false,
```

File: .eslintrc.js [hunk 24]
```diff
@@ -0,0 +1,230 @@
...
+        "no-dupe-class-members": "error",
+        "no-dupe-keys": "error",
+        "no-duplicate-case": "error",
+        "no-empty-character-class": "error",
+        "no-empty-pattern": "error",
+        "no-eval": "error",
+        "no-ex-assign": "error",
+        "no-extend-native": "error",
+        "no-extra-bind": "error",
+        "no-extra-boolean-cast": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion
        "no-extra-boolean-cast": "off",
```

File: .eslintrc.js [hunk 25]
```diff
@@ -0,0 +1,230 @@
...
+        "no-delete-var": "error",
+        "no-dupe-args": "error",
+        "no-dupe-class-members": "error",
+        "no-dupe-keys": "error",
+        "no-duplicate-case": "error",
+        "no-empty-character-class": "error",
+        "no-empty-pattern": "error",
+        "no-eval": "error",
+        "no-ex-assign": "error",
+        "no-extend-native": "error",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

Comment (danilmoro3) [reply .eslintrc.js]: `disallow extending native types`

File: .eslintrc.js (hunk 24 above)
Comment (danilmoro3) [reply .eslintrc.js]: `disallow unnecessary boolean casts`

File: .eslintrc.js (hunk 5 above)
Comment (danilmoro3) [reply .eslintrc.js]: `disallow 'new' operators outside of assignments or comparisons`

File: .eslintrc.js [hunk 26]
```diff
@@ -0,0 +1,230 @@
...
+        "comma-spacing": ["error", { "before": false, "after": true }],
+        "comma-style": ["error", "last"],
+        "constructor-super": "error",
+        "dot-location": ["error", "property"],
+        "eol-last": ["error", "always"],
+        "eqeqeq": ["error", "always", { "null": "ignore" }],
+        "func-call-spacing": ["error", "never"],
+        "generator-star-spacing": ["error", { "before": false, "after": true }],
+        "handle-callback-err": ["error", "^(err|error)$" ],
+        "indent": ["error", 4],
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion
        "indent": ["error", 4, { "SwitchCase": 1 }],
```

File: .eslintrc.js [hunk 27]
```diff
@@ -0,0 +1,230 @@
...
+        "SprykerAjax": "readonly",
+        "SprykerAlert": "readonly",
+        "SprykerAjaxCallbacks": "readonly",
+        "confirm": "readonly",
+        "alert": "readonly",
+        "localStorage": "readonly",
+        "antelope": "readonly",
+        "Event": "readonly",
+        "CustomEvent": "readonly",
+        "Image": "readonly",
```
Comment (danilmoro3) [inline .eslintrc.js]:
```suggestion

```

File: lib/eslint.js [hunk 28]
```diff
@@ -0,0 +1,29 @@
+const CLIEngine = require("eslint").CLIEngine;
+const globby = require("globby");
+const settings = require('./settings');
+
+const getPathPattern = async () => {
+    const pathPattern = [`${settings.paths.bundlesDir}/**/*.js`];
+    const pathPatternEco = `${settings.paths.ecoDir}/**/*.js`;
+    const ecoFilesToVerify = await globby([pathPatternEco]);
```
Comment (tolerants) [inline lib/eslint.js:8]: We do we need to verify it on this level?

**Review by tolerants — CHANGES_REQUESTED** [review]

File: lib/eslint.js (hunk 28 above)
Comment (zamlely) [reply lib/eslint.js:8]: Yes, it's needed. ESlint crashes if gets undefined path to linting and in this case we should to check existing eco modules before.

**Review by tolerants — APPROVED** [review]

**Pull request merged**